unit/solvers/refinement/bv_refinement/add_lemma.cpp

```diff
@@ -0,0 +1,136 @@
+/*******************************************************************\
+
+ Module: Unit tests for bv refinement
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+
+#include <util/arith_tools.h>
+#include <util/std_types.h>
+#include <util/std_expr.h>
+#include <solvers/refinement/bv_refinement.h>
+#include <solvers/sat/satcheck.h>
+#include <util/symbol_table.h>
+#include <java_bytecode/java_types.h>
+#include <util/config.h>
+#include <iostream>
+
+SCENARIO("add lemma", "[core][solvers][refinement][bv_refinement]")
+{
+  // Without this line, the program aborts with:
+  // `Reason: pointer must have non-zero width`
+  config.set_arch("none");
+
+  GIVEN("A solver")
+  {
+    satcheck_no_simplifiert sat_check;
+    symbol_tablet symbol_table;
+    namespacet ns(symbol_table);
+    bv_refinementt::infot info;
+    info.ns = &ns;
+    info.prop = &sat_check;
+    info.refine_arithmetic = true;
+    info.refine_arrays = true;
+    info.max_node_refinement = 5;
+    info.ui = ui_message_handlert::uit::PLAIN;
+    bv_refinementt solver(info);
+
+    WHEN("We give two contradictory inputs")
+    {
+      const symbol_exprt s1("symbol_var1", java_int_type());
```

Comment:
Does this need to be tied to the java types?

Reply:
Not really, I could replace it by `signedbv_typet(32)` but I find `java_int_type()` clearer. If it's preferable to remove dependencies on java_bytecode I will switch.

Reply:
I would prefer less dependencies.  Whether that is globally preferable ...   One advantage could be that you could also test unsigned and different bit lengths.

Reply:
Given the recent move to separate out the Java front-end I'd really prefer this.